### PR TITLE
Add emoji plugin

### PR DESCRIPTION
**Pull request by tyilo**
Works now with 0.18.

However it seems like albert doesn't show the correct icon, but picks one and caches it?
image

Any ideas @ManuelSchneid3r ?

### CODE REVIEW

**Comment by ManuelSchneid3r** [comment]
Does the plugin really deliver different paths? Use info(...) to check.

**Comment by tyilo** [comment]
> Does the plugin really deliver different paths? Use info(...) to check.

Yes it does.

File: emoji/__init__.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    def handleQuery(self, query):
+        query_tokens = query.string.strip().lower().split()
+        if not query_tokens:
+            return
+
+        for emoji in self.matched_emojis(query_tokens):
+            query.add(
+                Item(
+                    id="emoji",
```
Comment (ManuelSchneid3r) [inline emoji/__init__.py:136]: Different items have to have different ids

**Comment by tyilo** [comment]
I have fixed it now, however `albert` seems to segfault pretty easily:
```
01:31:16 [info:albert] Loading plugin emoji
01:31:16 [debg:albert] [76ms] Successfully loaded 'emoji'
01:31:16 [info:albert] Loading plugin api_test
01:31:16 [info:python] initialize
01:31:16 [debg:albert] [1ms] Successfully loaded 'api_test'
01:31:16 [info:albert] Loading plugin python_eval
01:31:16 [debg:albert] [0ms] Successfully loaded 'python_eval'
01:31:16 [debg:albert] [123ms] Successfully loaded 'python'
01:31:16 [debg:albert] TIME:     22 µs ['globalsearch':'']
01:31:21 [debg:albert] Query deleted. [#0 '']
01:31:21 [debg:albert] TIME:    227 µs ['albert':':']
01:31:21 [debg:albert] TIME:    251 µs ['system':':']
01:31:21 [debg:albert] TIME:    136 µs ['calculator':':']
01:31:21 [debg:albert] TIME:    299 µs ['plugins':':']
01:31:21 [debg:albert] TIME:    776 µs ['applications':':']
01:31:21 [debg:albert] TIME: 281953 µs ['files':':']
01:31:21 [debg:albert] TIME: 287669 µs ['globalsearch':':']
01:31:22 [debg:albert] Query deleted. [#1 ':']
01:31:22 [debg:albert] TIME:     23 µs ['albert.emoji':'']
01:31:22 [debg:albert] TIME:   3925 µs ['albert.emoji':'b']
fish: Job 1, 'albert --debug' terminated by signal SIGSEGV (Address boundary error)
```

**Comment by ManuelSchneid3r** [comment]
I hate those segfaults. Damnit. The docker extension has those too. i already spent hours on this one and gave up. prokrastinating this one although pretty severe.

**Comment by ManuelSchneid3r** [comment]
https://termbin.com/evni got this on macos. convert is installed

**Comment by ManuelSchneid3r** [comment]
```shell
manuel::xenon ~ identify -list format | grep -Ei "Description|pango"
   Format  Module    Mode  Description
    PANGO* PANGO     ---   Pango Markup Language
```

**Comment by stevenxxiu** [comment]
I might have encountered a similar segfault. In my case, if I give *Albert Launcher* too many results it appears to segfault more often. My code's at [albert_unicode/\_\_init__.py at master · stevenxxiu/albert_unicode](https://github.com/stevenxxiu/albert_unicode/blob/master/__init__.py#L20).

**Comment by ManuelSchneid3r** [comment]
Do your items all have unique ids? This has often been an issue in the past

**Comment by stevenxxiu** [comment]
> Do your items all have unique ids? This has often been an issue in the past

If you're talking to me, they should have unique IDs. I'm using the character as a part of the ID. They should be different.

**Comment by ManuelSchneid3r** [comment]
The sedfaults should be fixed. everthing fine @tyilo

**Comment by stevenxxiu** [comment]
> The sedfaults should be fixed. everthing fine @tyilo

Thanks that fixes it for my plugin too.

**Comment by BarrensZeppelin** [comment]
I can't trigger any segfaults. I think the plugin works as intended.

**Comment by tomsquest** [comment]
Working fine here.

@tyilo the icon is displayed "twice" (the action icon, and in the text). Do you think it is useful?

image

**Comment by ManuelSchneid3r** [comment]
Well there are indeed some things to complain, but for now i'll merge it. We can progress from there on.

**Comment by tyilo** [comment]
> @tyilo the icon is displayed "twice" (the action icon, and in the text). Do you think it is useful?

I think the screenshot by @ManuelSchneid3r shows why this is needed.

**Comment by ManuelSchneid3r** [comment]
these glitches probably stem from the images not being there. a former emoji approach simply build all the icons on initialize.

**Comment by ManuelSchneid3r** [comment]
this one https://github.com/albertlauncher/python/blob/master/.archive/unicode_emoji/__init__.py

**Pull request merged**